rskj-core/src/main/java/co/rsk/peg/federation/constants/FederationRegTestConstants.java

```diff
@@ -56,4 +58,26 @@ public FederationRegTestConstants(List<BtcECKey> federationPublicKeys) {
         // e1b17fcd0ef1942465eee61b20561b16750191143d365e71de08b33dd84a9788
         oldFederationAddress = "2N7ZgQyhFKm17RbaLqygYbS7KLrQfapyZzu";
     }
+
+    public static synchronized FederationRegTestConstants getInstance(List<BtcECKey> federationPublicKeys) {
+        if (federationPublicKeys == null || federationPublicKeys.isEmpty()) {
+            throw new IllegalArgumentException("Federation regtest public keys must not be null or empty.");
+        }
+
+        if (instance == null) {
+            instance = new FederationRegTestConstants(federationPublicKeys);
+        } else if (!federationPublicKeys.equals(instance.getGenesisFederationPublicKeys())) {
+            throw new IllegalStateException("Federation regtest constants is already initialized with different keys.");
+        }
+
+        return instance;
+    }
+
+    public static synchronized FederationRegTestConstants getInstance() {
```

Comment:
Should we have javadoc on these methods? For example, this one throws an exception when regtest constants is not initialized first. With a javadoc, while calling this method, it would help develpers quickly see that this one throws an exception if the regtest constants is not yet initialized.

Reply:
Agree, added the javadocs.